### PR TITLE
Fix:empty segments crashed the new version of XTreeGetRecord

### PR DESCRIPTION
**Pull request by GabrieleManduchi**
The crash occurred in Squish section. A check has been added before in order to pass only non empty segments to Squish function.

### CODE REVIEW

File: xtreeshr/XTreeGetTimedRecord.c [hunk 1]
```diff
@@ -341,7 +341,11 @@ EXPORT int XTreeGetTimedRecord(int inNid, mdsdsc_t *startD,
 
     MdsFree1Dx(&dataXds[currSegIdx], 0);
     MdsFree1Dx(&dimensionXds[currSegIdx], 0);
-    signals[currSegIdx] = (mds_signal_t *)resampledXds[currSegIdx].pointer;
+//After resampling some segments may be empty: consider only nonempty segments
+    if(resampledXds[currSegIdx].l_length > 0)
+      signals[nonEmptySegIdx++] = (mds_signal_t *)resampledXds[currSegIdx].pointer;
+    else //Keep track that the segment is not considered
+      signalsApd.arsize -= sizeof(mds_signal_t *);
```
Comment (zack-vii) [inline xtreeshr/XTreeGetTimedRecord.c]: shoukd it not better be using the length field of signalsApd instead of sizeof. best would be though to calculate arsize based on nonEmptySegIdx an length after the for loop.

Comment (zack-vii) [reply xtreeshr/XTreeGetTimedRecord.c]: we should add some tests that tackle those kind of edge cases because xtreeshr is used frequently and its a pitty its not tested well. i may be able to contribute. we should add these tests to treeUnitTest where i test the timecontext stuff. maybe there is a lightweight example that would trigger the issue w/o preexisting tree files.

**Review by zack-vii — CHANGES_REQUESTED** [review]
the arsize should be calculated at the end of the loop.
and we need tests. but the can follow in a different pr

File: xtreeshr/XTreeGetTimedRecord.c (hunk 1 above)
Comment (GabrieleManduchi) [reply xtreeshr/XTreeGetTimedRecord.c]: Ok, removed length update after the loop

**Review by zack-vii — APPROVED** [review]
